server/master/src/main/java/org/apache/accumulo/master/Master.java

```diff
@@ -1336,7 +1332,11 @@ public void assignedTablet(KeyExtent extent) {
   }
 
   @Override
-  public Collection<KeyExtent> migrations() {
-    return migrations.keySet();
+  public Set<KeyExtent> migrations() {
+    Set<KeyExtent> migrationsCopy = new HashSet<KeyExtent>();
+    synchronized (migrations) {
+      migrationsCopy.addAll(migrations.keySet());
+    }
+    return migrationsCopy;
```

Comment:
I'll go ahead and do the immutability, as well as rename the method... but I'll add a Javadoc to clearly state that it's immutible. It looks like the other (test) implementations are already immutable anyway.